backend/dal/Helpers/Extensions/PropertyExtensions.cs

```diff
@@ -135,5 +146,54 @@ public static void ThrowIfPropertyNotInProjectAgency(this Entity.Property parcel
             if (!parcel.AgencyId.HasValue || !projectAgencyIds.Contains(parcel.AgencyId.Value))
                 throw new InvalidOperationException("Properties may not be added to Projects with a different agency.");
         }
+
+        public static void AttachProjectsStatus(this PimsContext context, Property[] properties)
+        {
+            try
+            {
+                properties = properties.Where(x => !String.IsNullOrWhiteSpace(x.ProjectNumber)).ToArray();
+                if (properties.Length > 0)
+                {
+                    var projectNumbers = properties
+                        .Select(x => x.ProjectNumber).ToArray();
+                    var propertyProjectStatuses = context.Projects.Where(x =>
+                            !String.IsNullOrWhiteSpace(x.ProjectNumber) && projectNumbers.Contains(x.ProjectNumber))
+                        .Select(p =>
+                            new {
+                                Id = p.StatusId,
+                                ProjectNumber = p.ProjectNumber
+                            })
+                        .ToArray();
+
+                    var statuses = context.ProjectStatus.
+                        Where(ps => propertyProjectStatuses.Select(x => x.Id).Contains(ps.Id))
+                        .Select(s => new {Id = s.Id, Code = s.Code})
+                        .ToArray();
+
+                    properties.ForEach(property =>
+                    {
+                        if (!String.IsNullOrWhiteSpace(property.ProjectNumber))
+                        {
+                            var propertyProjectStatus = propertyProjectStatuses
```

Comment:
If all we want is to get the property project status you can change this whole function.  Our business rule is a property can only be associated to a single **active** project.  Since I believe the primary purpose of this is for the ERP statuses, then just update the `Property` to include a collection association to `Projects`.

Something like the following.  This is just seudo-code.
```
this.Context.Properties(p => p.Projects.Any(pr => statusIds.Contains(pr.StatusId))).Select(p => new { Property = p, Status = p.Projects.FirstOrDefault(pr => statusIds.Contains(pr.StatusId)) })
```